jkube-kit/enricher/api/src/main/java/org/eclipse/jkube/kit/enricher/api/BaseEnricher.java

```diff
@@ -148,29 +149,37 @@ protected Long getOpenshiftDeployTimeoutInSeconds(Long defaultValue) {
     /**
      * This method overrides the controller name value by the value provided in XML config.
      *
-     * @param resourceConfig resource config from plugin configuration
+     * @param controllerResourceConfig Controller Resource config from plugin configuration
      * @param defaultValue default value
      * @return string as controller name
      */
-    protected String getControllerName(ResourceConfig resourceConfig, String defaultValue) {
-        return Optional.ofNullable(resourceConfig).map(ResourceConfig::getControllerName).orElse(defaultValue);
+    protected String getControllerName(ControllerResourceConfig controllerResourceConfig, String defaultValue) {
+        return Optional.ofNullable(controllerResourceConfig).map(ControllerResourceConfig::getControllerName).orElse(defaultValue);
+    }
+
+    protected ControllerResourceConfig getControllerResourceConfig() {
+        ResourceConfig resourceConfig = getConfiguration().getResource();
+        if (resourceConfig != null && resourceConfig.getController() != null) {
+            return resourceConfig.getController();
+        }
+        return ControllerResourceConfig.builder().build();
```

Comment:
Related to the previous comment, this method should call `getControllers`

If there is more than one controller, then we should warn the user.